src/server/node_manager.c

```diff
@@ -7114,11 +7114,27 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
-                        if (svr_init == FALSE && (pnode->nd_state & (INUSE_DOWN | INUSE_STALE))) {
-                                   free(phowl);
-                                   free(execvncopy);
-                                   return (PBSE_BAD_NODE_STATE);
-                        }
+                	if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE))&& (svr_init == FALSE)) {
+                            if (objtype == JOB_OBJECT) {
+				free(phowl);
+                                free(execvncopy);
+                                return (PBSE_BAD_NODE_STATE);
+			    }
+			    else if(presv->ri_wattr[RESV_ATR_resv_standing].at_val.at_long) {
+				/* Settin Standing reservation to Degraded */
+				resv_setResvState(presv, RESV_DEGRADED, RESV_DEGRADED);
+			    }
+			    else {
+				int est_retryTime;
+				est_retryTime = time_now+ 10; /* Time for Normal reservation to retry, value in Seconds */
+				
+				set_resv_retry(presv, est_retryTime);
+				free(phowl);
+                                free(execvncopy);
+				return (PBSE_BAD_NODE_STATE);
```

Comment:
This is not right.  If you reject the set_nodes() call, the confirm will fail.  This means it will attempt to be reconfirmed when it is not degraded.  The more I think about this, I think we should treat all reservations the same.  Just mark them degraded and let the degraded reservation machinery handle reconfirming it.